scripts/solve_network.py

```diff
@@ -1055,14 +1055,14 @@ def extra_functionality(n, snapshots):
     if config["sector"]["enhanced_geothermal"]["enable"]:
         add_flexible_egs_constraint(n)
 
-    if snakemake.params.custom_extra_functionality:
-        source_path = snakemake.params.custom_extra_functionality
+    if params.custom_extra_functionality:
+        source_path = params.custom_extra_functionality
         assert os.path.exists(source_path), f"{source_path} does not exist"
         sys.path.append(os.path.dirname(source_path))
         module_name = os.path.splitext(os.path.basename(source_path))[0]
         module = importlib.import_module(module_name)
         custom_extra_functionality = getattr(module, module_name)
-        custom_extra_functionality(n, snapshots, snakemake)
+        custom_extra_functionality(n, snapshots)
 
 
 def solve_network(n, config, solving, **kwargs):
```

Comment:
I think, since you assume `params` is there in the function, it must be a compulsory argument:
```suggestion
def solve_network(n, config, params, solving, **kwargs):
```

All calls to `solve_network()` in PyPSA-Eur must be adjusted like done in `solve_operations_networks.py`.

```py
n = solve_network(
    n,
    config=snakemake.config,
    params=snakemake.params,
    ...
)